wordpress-mobile/WordPress-Android
# Post change conflict resolution dialog

This is a follow up to PR #8940 

Comes from https://github.com/wordpress-mobile/WordPress-Android/issues/5984#issuecomment-453706747 

After implementing #8940 as a fix to https://github.com/wordpress-mobile/WordPress-Android/issues/8932, posts with local changes are not updated with the remote, and there is no indication if the remote had changes as well.

Quoting @megsfulton  here for completeness:

> One approach we could take that would be minimally disruptive but still communicate that there's an issue to be resolved: 

> - When the post list is fetched, display a line on the post card indicating that there's a conflict. Similar to how "Local Changes" appears but with some sort of warning icon.
> - When the user opens the post ask them how they would like to resolve the conflict. Until they open the post to take an action both conflicted states would continue to exist. 

What this PR does is:
- [x] show a new status that tells the user if a Post has a conflict (needs design review, it only shows `Conflicted` for now).
- [x] when the user taps on a Post with  a `Conflicted` status, it shows a dialog asking whether they would like to load and the Post from the remote, or edit the local copy. They can cancel and not take any action by dismissing the dialog (either tap back arrow key or tap on the screen outside the dialog area)

Corresponding FluxC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1077

*Note:* this needs copy review, the text and approach here does not intend to be final but to show what the functionality would feel like. Also, the new status's color and icon should probably be different than now? cc @megsfulton .

To test:
CASE A: (show the new status on the Posts list)
1. Create a draft while I am online in the app. That'll create the draft in remote.
2. Go offline
3. Make changes to the post and go back. This will save the changes as local draft
4. Go to web or any other client and make changes to the post.
5. Publish the post.
6. on the app, you should be on the posts list. turn airplane mode off
7. pull to refresh.
8. you should see now the post has a "Conflicted" state.

<img width="330" alt="screen shot 2019-01-16 at 12 30 15" src="https://user-images.githubusercontent.com/6597771/51270426-9dd73080-19a3-11e9-93e5-fdab461462d0.png">


Dialog shown here:
<img width="336" alt="screen shot 2019-01-16 at 12 30 25" src="https://user-images.githubusercontent.com/6597771/51270444-af203d00-19a3-11e9-9bb5-66c8cf5b9830.png">


CASE B: (simple conflict resolution dialog - edit local)
1. follow steps 1 to 8 above.
2. now tap on the post
3. a new dialog should be shown, offering 2 options: `EDIT LOCAL` and `LOAD REMOTE`
4. if you tap EDIT LOCAL, the local version is loaded. Beware once you tap on it, you've made a choice, but the contents are not lost (these are added to the revision history).
5. if you tap LOAD REMOTE, an attempt to load the latest version from remote is made. You'll need to be online for this (this prerequisite is checked for, and otherwise a toast is shown). You'll see how it gets updated.
![editlocal](https://user-images.githubusercontent.com/6597771/51270455-bb0bff00-19a3-11e9-9a8e-1948fd57abc4.gif)

CASE C: (simple conflict resolution dialog - load remote)
1. follow steps 1 to 8 in case A above.
2. now tap on the post
3. a new dialog should be shown, offering 2 options: `EDIT LOCAL` and `LOAD REMOTE`
4. if you tap LOAD REMOTE, an attempt to load the latest version from remote is made. You'll need to be online for this (this prerequisite is checked for, and otherwise a toast is shown). You'll see how it gets updated.
![loadremote](https://user-images.githubusercontent.com/6597771/51270461-be9f8600-19a3-11e9-82e0-61e6654716f7.gif)

Update release notes:
